### PR TITLE
Base Extension

### LINKED ISSUE #1: Create browser plugin
This new plugin could simulate (or even directly include) the original integration JS, that finds `mailto` and `tel` links and opens popup with loaded custom LiveAgent server-rendered page.

- existing popup JS implementation is here: https://github.com/QualityUnit/LiveAgent/pull/5628/files#diff-ceef29ff966e447e5af40cbea526f4c4e1e79b8ace592b85f3d6a21a9eb7eb79 - after load it could be executed like: 

```JS
            LARemoteActions.create("<LiveAgent-URL>agent/remote_actions.php", true);
```

The popup is created using https://atomiks.github.io/tippyjs/ and the page content is loaded into ShadowDOM. 

- `LiveAgent-URL` should be the only user configuration needed directly in the plugin for now (later we could add more settings like theme, website whitelist, blacklist etc...)

- the template inside the popup requires font `<LiveAgent-URL>/themes/agent/material/fonts/la-icon-font.woff2` included as `la-icon-font` - but the only way how I got it working is by loading it from outside - inside ShadowDOM I was unable to load it.

- plugin could be simple enough to be compatible with both - Firefox and Chrome simultaneously

### PR DESCRIPTION
**Pull request by MagicRB**
Outstanding issues:

- a WebExtension must specify which sites, aside from the one it's running on, it wants to access, according to [MDN](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions#host_permissions). We could specify `<all_urls>`, according to [MDN](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Match_patterns#%3Call_urls%3E), but I'd like to figure out a better way, even though I doubt it's existence
- I couldn't get the code to actually do anything else than display `Loading...`, I probably got the address wrong. Is there a LiveAgent instance I could test it on?

Todo:

- Limit on which sites the extension runs
- Allow the user to specify the URL to their LiveAgent instance
- Allow the user to pick a theme, dark mode rocks!

My questions:

- did I unnecessarily complicate it by using webpack?  

Close https://github.com/QualityUnit/LiveAgent-remote-actions/issues/1

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,298 @@
+
```
Comment (martincivan) [inline .gitignore:1]: this is really exhausting gitignore for such a small thing - I would expect  just `.idea/` ... but whateever :shrug:

File: README.org [hunk 2]
```diff
@@ -0,0 +1 @@
+* LiveAgent Remote Actions
```
Comment (martincivan) [inline README.org]: Why do we need another README file?

File: flake.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
+{
```
Comment (martincivan) [inline flake.nix:1]: I dont think that nix is very good choice - many developers here use windows and even on linux it is not included in many package managers... I would say that https://github.com/mozilla/web-ext would be more appropriate for packaging this...

File: manifest.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
+    "manifest_version": 2,
+    "name": "LiveAgent-remote-actions",
+    "version": "1.0",
+
+    "description": "AAAAAAAA",
```
Comment (martincivan) [inline manifest.json:6]: This is not very nice description.

File: manifest.json [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    },
+
+    "content_scripts": [
+	{
+	    "matches": ["<all_urls>"],
+	    "js": ["dist/bundle.js"]
+	}
+    ],
+    "permissions": [
+	"*://liveagent.com/"
```
Comment (martincivan) [inline manifest.json]: LiveAgent is not always on this domain.

File: src/main.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
+import tippy from 'tippy.js';
+import 'tippy.js/themes/light.css';
+import './tooltip.css';
+
+const mailtoParser = require("mailto-parser");
+
+console.log("asd");
```
Comment (martincivan) [inline src/main.js]: forgotten debug logs

File: src/main.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+				})
+			).text();
+		    instance.setContent(elem);
+		},
+	    });
+	}
+    }
+};
+
+window.LARemoteActions.create("https://liveagent.com/agent/remote_actions.php", false);
```
Comment (martincivan) [inline src/main.js]: This should be user-configurable.

**Review by martincivan — COMMENTED** [review]
@MagicRB 
> a

I'd say, that this may be solved by allowing CORS in HTTP headers - try it on your PC as I wrote you in Slack.

> b

That PR is not yet deployed at all due to bug in our infrastructure that is unable to deploy versions from PRs... It is reported and we are waiting for our server admins to fix it.

But you could run simulate it locally - you could install LIveAgent at your PC directly or just mock it with some  simple script...

Limit on which sites the extension runs - :+1: 
Allow the user to specify the URL to their LiveAgent instance - :+1: 
Allow the user to pick a theme, dark mode rocks! - sure, but not yet - will be in v.1.1.


>    did I unnecessarily complicate it by using webpack?

Webpack is IMO fine, but I dont think that anyone here would like to bother installing Nix. 
So I would say that webpack (or/and https://github.com/mozilla/web-ext) should be fine. 
Also I am not sure if we need to babelify the JS - all the code should work fine in all modern browsers, but whatever, not a big deal...

**Comment by MagicRB** [comment]
The nix bit is just for me, simplifies packaging a bit, you guys dont have to use it all, though I highly encourage it :D

File: flake.nix (hunk 3 above)
Comment (MagicRB) [reply flake.nix:1]: of course, like I said Nix is there just to ease the development for me, no one else has to to touch it

File: .gitignore (hunk 1 above)
Comment (MagicRB) [reply .gitignore:1]: I included everything I could think of, just to be sure that someone doesn't commit anything by accident

File: manifest.json (hunk 4 above)
Comment (MagicRB) [reply manifest.json:6]: I'm not really good at describing things. 
image

File: manifest.json (hunk 5 above)
Comment (MagicRB) [reply manifest.json]: Fixed, now the extensions asks for `<all_urls>` and the filtering is done manually in the extension with regexes

File: src/main.js (hunk 6 above)
Comment (MagicRB) [reply src/main.js]: Removed

File: src/main.js (hunk 7 above)
Comment (MagicRB) [reply src/main.js]: Done

**Review by martincivan — APPROVED** [review]
@MagicRB 
Alright looks fine, please clean all those debug logs and we can merge this.

**Pull request merged**